10up/distributor
# Tweak: create pot file with WP CLI and  move translations to JS files

### Description of the Change



 I create a bash script that uses `package.json --> files` to create a pot file for the addon. I implemented `EnqueueScript`, which manages script dependencies and versions based on the asset file generated by `@wordpress/dependency-extraction-webpack-plugin` package.
 
Closes #947 #267 #660

### How to test the Change

- Run `npm run makepot` command. It should generate a translation file that contains translations from a javascript file (only in `dist/js` directory). 
- Each script and style file should load correctly, which their versions and dependencies mentioned in the assets file.
- Each feature depends upon javascript and should work as expected.
- You should be able to edit translatable strings in Javascript or PHP files. You can use [Loco translate](https://wordpress.org/plugins/loco-translate/) plugin for testing.
  |  Action | Screenshot |
  |------------|--------|
  | Edit string | <img width="987" alt="image" src="https://user-images.githubusercontent.com/1784821/223175616-081745b7-ecd2-49dc-a43e-da0529b27e12.png"> <img width="847" alt="image" src="https://user-images.githubusercontent.com/1784821/223176688-a0835afb-91fd-4dbc-945c-7bb72f608c2a.png"> |
  | View edited string | <img width="1101" alt="Screenshot 2023-03-06 at 10 14 52 PM" src="https://user-images.githubusercontent.com/1784821/223175750-498627f6-a8f3-4e0f-affa-61e97a943f73.png"> |
  | Translation files in plugin directory | <img width="485" alt="image" src="https://user-images.githubusercontent.com/1784821/223176866-4569fadd-95c1-49e8-9239-5ae763fcf33d.png"> |

### Changelog Entry

> Added - Generate POT file with WP-CLI.
> Added - Register scripts with `EnqueueScript` class.

## Notable Changes
- Rename `./tests/php/bootstrap.php.dist --> ./tests/php/bootstrap.php` to get code editor support.

## Todo before release
- [ ] Ensure WP CLI exists in the built environment before generating the pot file. 

### Checklist:
